### PR TITLE
Type num

### PR DESCRIPTION
**Pull request by tobyink**
Adds a Num type like Int, but allows decimals and scientific notation (like `-1.5e6`). Anything that `looks_like_number` allows.

Right now I'm having to just use the String type for non-integer numbers, so having a Num type would be good!

### CODE REVIEW

**Comment by tobyink** [comment]
Oh yes, I forgot to mention I can count to five in Japanese. I know hello, goodbye, and thank you. That's about all the Japanese I know, even though I was born in Tokyo and I had Japanese lessons as a child.

So what I mean to say is that the Japanese portions of the documentation need fixing by someone who isn't me!

**Comment by sago35** [comment]
> So what I mean to say is that the Japanese portions of the documentation need fixing by someone who isn't me!

I would like to make my comments in English, not Japanese.
I live in Japan and my English is not very good.
I can read English for the most part, but it takes a long time to write in English.
I translated most of the Japanese into English before releasing Getopt::Kingpin, but it looks like it was left over.

I'd appreciate it if you could write this PR in English.

File: lib/Getopt/Kingpin/Type/Num.pm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Getopt::Kingpin::Type::Num は、Getopt::Kingpin内で使用する型定義です。
+
+=head1 METHOD
+
+=head2 set_value($value)
+
+$self->valueに値を設定します。
+値はqr/-?[0-9]+$/で定義されるもののみです。
+
```
Comment (sago35) [inline lib/Getopt/Kingpin/Type/Num.pm]: I want you to write it in English.
Thank you.

**Comment by sago35** [comment]
I think it's very good.
The int() is pulled from Go's kingpin, but it's certainly very restrictive.

File: t/28_num.t [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    push @ARGV, qw(--x 0);
+
+    my $kingpin = Getopt::Kingpin->new;
+    my $x = $kingpin->flag('x', 'set x')->num();
+    $kingpin->parse;
+
+    is $x->value, 0;
+    ok $x == 0;
+};
+
```
Comment (sago35) [inline t/28_num.t:19]: Please add a test for `looks_like_number` cases.
For example, `1e3` or `-1.5e6`.

**Comment by tobyink** [comment]
If you like, I can try translating other parts of the documentation.

**Comment by sago35** [comment]
> If you like, I can try translating other parts of the documentation.

Please.
Thank you!

File: lib/Getopt/Kingpin/Type/Num.pm (hunk 1 above)
Comment (sago35) [reply lib/Getopt/Kingpin/Type/Num.pm]: 75dcfd49876ce1dedca9f0bf9dcbadce7c5759e1

**Comment by sago35** [comment]
LGTM
Thank you for your great contribution.

**Pull request merged**